File: Lonely-Integer.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'lonelyinteger' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY a as parameter.
     */

    public static int lonelyinteger(List<int> a)
    {
        int uniqueElement = 0;

        foreach (int num in a)
        {
            uniqueElement ^= num;
        }

        return uniqueElement;
    }    


    public static int lonelyinteger2(List<int> a)
    {
        Dictionary<int, int> frequencyMap = new Dictionary<int, int>();

        foreach (int num in a)
        {
            if (frequencyMap.ContainsKey(num))
                frequencyMap[num]++;
            else
                frequencyMap[num] = 1;
        }

        foreach (var pair in frequencyMap)
        {
            if (pair.Value == 1)
            {
                return pair.Key;
            }
        }

        return -1;
    }

}

class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<int> a = Console.ReadLine().TrimEnd().Split(' ').ToList().Select(aTemp => Convert.ToInt32(aTemp)).ToList();

        int result = Result.lonelyinteger(a);

        textWriter.WriteLine(result);

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Lonely-Integer.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'lonelyinteger' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts INTEGER_ARRAY a as parameter.
     */


    public static int lonelyinteger(List<Integer> a) {
        int uniqueElement = 0;

        for (int num : a) {
            uniqueElement ^= num;
        }

        return uniqueElement;
    }


    public static int lonelyinteger2(List<Integer> a) {
        Map<Integer, Integer> frequencyMap = new HashMap<>();

        for (int num : a) {
            frequencyMap.put(num, frequencyMap.getOrDefault(num, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            if (entry.getValue() == 1) {
                return entry.getKey();
            }
        }

        return -1;
    }
    

    public static int lonelyinteger3(List<Integer> a) {
        int n = a.size();

        Set<Integer> uniqueElements = new HashSet<>(a);
        int u = uniqueElements.size();

        if (n == 1) return a.get(0);
        if (u <= 1) return -1;

        int k = (n - 1) / (u - 1);

        int result = 0;

        for (int i = 0; i < 32; i++) {
            int bitSum = 0;

            for (int num : a) {
                if (((num >> i) & 1) == 1) {
                    bitSum++;
                }
            }

            if (bitSum % k != 0) {
                result |= (1 << i);
            }
        }

        return result;
    }    
}


public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> a = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
            .map(Integer::parseInt)
            .collect(toList());

        int result = Result.lonelyinteger(a);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
